lib/src/metta/runner/stdlib/core.rs

```diff
@@ -407,6 +346,18 @@ mod tests {
         assert_eq!(result, Ok(vec![vec![expr!("nok")]]));
     }
 
+    #[test]
+    fn metta_case_error() {
+        let program = "
+            (= (err) (Error (err) \"Test error\"))
+            !(assertEqual (case (err) (
+              ((Error $a \"Test error\") ok)
+              ($_ (Error $_ \"Error is expected\"))
+              )) ok)
+        ";
+        assert_eq!(run_program(program), Ok(vec![vec![UNIT_ATOM]]));
```

Comment:
I would not use MeTTa "assert" in Rust unit tests if it is possible. It makes test dependent on "assert" behavior. 
```suggestion
            (= (err) (Error (err) \"Test error\"))
            !(case (err) (
              ((Error $a \"Test error\") ())
              ($other (Error $other \"Error is expected\"))
              ))";
        assert_eq!(run_program(program), Ok(vec![vec![UNIT_ATOM]]));
```